kratos/utilities/geometry_tester.cpp

```diff
@@ -944,8 +961,74 @@ bool GeometryTesterUtility::VerfiyShapeFunctionsValues(
 /***********************************************************************************/
 /***********************************************************************************/
 
+bool GeometryTesterUtility::VerifyShapeFunctionsSecondDerivativesValues(
+    GeometryType& rGeometry,
+    GeometryType::CoordinatesArrayType& rGlobalCoordinates,
+    std::stringstream& rErrorMessage
+    )
+{
+    GeometryType::CoordinatesArrayType local_coordinates;
+    rGeometry.PointLocalCoordinates( local_coordinates, rGlobalCoordinates );
+
+    Vector f_1 = ZeroVector(rGeometry.size());
+    Vector f_2 = ZeroVector(rGeometry.size());
+    Vector f_3 = ZeroVector(rGeometry.size());
+    Vector f_4 = ZeroVector(rGeometry.size());
+
+    DenseVector<Matrix> DDN_DX, H;
+    Vector ei,ej,f;
+
+    double delta = 1e-1;
+    unsigned int dim = rGeometry.Dimension();
+    rGeometry.ShapeFunctionsSecondDerivatives(DDN_DX,local_coordinates);
+
+    if ( H.size() != rGeometry.size() )
+    {
+        // KLUDGE: While there is a bug in
+        // ublas vector resize, I have to put this beside resizing!!
+        DenseVector<Matrix> temp( rGeometry.size() );
+        H.swap( temp );
+    }
+
+
+    for ( unsigned int i = 0; i < rGeometry.size(); ++i ) H[i].resize(dim, dim, false);
+
+    for (unsigned int i = 0; i<dim;i++){
+        for (unsigned int j = 0; j<dim;j++){
+            ei = ZeroVector(dim);
+            ei[i] = 1.0;
+            ej = ZeroVector(dim);
+            ej[j] = 1.0;
+            std::transform(ei.begin(), ei.end(), ei.begin(), [delta](double &c){ return c*delta; });
+            std::transform(ej.begin(), ej.end(), ej.begin(), [delta](double &c){ return c*delta; });
+            rGeometry.ShapeFunctionsValues(f_1,local_coordinates + ei + ej);
```

Comment:
... and then you're trying to add the 0..2 size vectors to ```local_coordinates```, which is probably of size 3.

Reply:
That may be the point, I will try setting ej and ei as ZeroVector of size 3 instead of size dim and see whats happens.